pallets/parachain-staking/src/lib.rs

```diff
@@ -438,9 +442,50 @@ pub mod pallet {
 				weight = weight.saturating_add(T::OnNewRound::on_new_round(round.current));
 				// pay all stakers for T::RewardPaymentDelay rounds ago
 				weight = weight.saturating_add(Self::prepare_staking_payouts(round.current));
+
+				// select total collators
+				let collators = <SelectedCandidates<T>>::get();
+
+				// if collators length is below or eq to 66% of max_collators,
+				// we don't mark any other collator as offline
+				let max_collators = <TotalSelected<T>>::get();
+
+				// track collators length
+				let mut len_counter = collators.len();
+
+				// iter collators to check which of them must be marked as offline
+				for collator in collators {
+					if let Some(info) = <CandidateLastActive<T>>::get(&collator) {
+						if round.current.saturating_sub(info.last_round)
+							> T::MaxOfflineRounds::get() && len_counter * 3
```

Comment:
I believe this loop cannot break early, because candidates must also be inserted in the `else` branch. This is for the case when a collator produces 0 blocks.

Reply:
Is like Tomasz says, we can't break the loop, we have to iterate the entire collators list to initialize the storage for the ones that haven't produced blocks, so that we can track their behaviour and mark them offline if they don't produce blocks.